### PR TITLE
Adding troubleshooting section for Elastic Agent on Kubernetes and Kustomize

### PR DESCRIPTION
**Pull request by eedugon**
Closes https://github.com/elastic/kibana/issues/186382

I've added a new section on the troubleshooting document for Kubernetes related Elastic Agents.

### CODE REVIEW

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 1]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+
+The previous generates a local file named `elastic_agent_installation_complete_manifest.yaml`, which you can use for further analysis. It contains the complete set of resources required for the {agent} installation, including:
+
+* RBAC objects (`ServiceAccounts`, `Roles`, etc. )
+
+* `ConfigMaps` and `Secrets` for {agent} configuration.
+
+* {agent} Standalone deployed as a `DaemonSet`
+
+* https://github.com/kubernetes/kube-state-metrics[Kube-state-metrics] deployed as a `Deployment`.
```
Comment (bmorelli25) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:864]: I would remove the periods from all of these for consistency

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 2]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+For example, the following command, originally provided by Kibana for an {agent} Standalone installation, has been modified to redirect the output for troubleshooting purposes:
+
+[source,sh]
+----
+kubectl kustomize https://github.com/elastic/elastic-agent/deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-standalone\?ref\=v8.15.3 | sed -e 's/JUFQSV9LRVkl/ZDAyNnZaSUJ3eWIwSUlCT0duRGs6Q1JfYmJoVFRUQktoN2dXTkd0FNMtdw==/g' -e "s/%ES_HOST%/https:\/\/7a912e8674a34086eacd0e3d615e6048.us-west2.gcp.elastic-cloud.com:443/g" -e "s/%ONBOARDING_ID%/db687358-2c1f-4ec9-86e0-8f1baa4912ed/g" -e "s/\(docker.elastic.co\/beats\/elastic-agent:\).*$/\18.15.3/g" -e "/{CA_TRUSTED}/c\ " > elastic_agent_installation_complete_manifest.yaml
+----
+
+The previous generates a local file named `elastic_agent_installation_complete_manifest.yaml`, which you can use for further analysis. It contains the complete set of resources required for the {agent} installation, including:
+
+* RBAC objects (`ServiceAccounts`, `Roles`, etc. )
```
Comment (bmorelli25) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:858]:
```suggestion
* RBAC objects (`ServiceAccounts`, `Roles`, etc.)
```

**Review by bmorelli25 — COMMENTED** [review]
Great stuff so far! I have a few grammar recommendations for you.

**Review by kilfoyle — COMMENTED** [review]
This looks great! @eedugon I'll wait for you to address Brandon's comments and then I'll give it a review as well.

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 3]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+One way to check if `state_*` metrics are being delivered to Elasticsearch is to check the log lines with the `"Non-zero metrics in the last 30s"` message and check the values of the "state_*" metricsets within the line, with something like:
++
+[source,sh]
+----
+kubectl logs -n kube-system elastic-agent-xxxx | grep "Non-zero metrics" | grep "state_"
+----
++
+If the previous return something like `"state_pod":{"events":213,"success":213}` for all `state_*` metricsets, it means the metrics are being delivered.
++
+. As a last resort, if you believe none of the Pods is acting as a leader, you can try deleting the `lease`, thereby generating a new one:
```
[the remark this answers is not in the document]
Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:940]: What about... "deleting the `lease`, for force the generation of a new one"? Better than `to generate a new one`?

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 4]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+
+. Check resources status and ensure they are all `Running`
++
+[source,sh]
+----
+kubectl get pods -n kube-system | grep elastic
+kubectl get pods -n kube-system | grep kube-state-metrics
+----
+
+. Describe the Pods in case they are in `Pending` state:
```
[the remark this answers is not in the document]
Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:890]: Thanks! yes, that's what I wanted to say :)

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 5]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+
+Possible issues:
+
+* Connectivity, authorization, or authentication issues when connecting to Elasticsearch:
++
+Ensure the API Key and Elasticsearch destination endpoint used during the installation is correct and is reachable from within the Pods.
++
+In an already installed system, the API Key is stored in a `Secret` named `elastic-agent-creds-<hash>`, and the endpoint is configured in the `ConfigMap` `elastic-agent-configs-<hash>`.
+
+* Only missing cluster-level metrics (provided by `kube-state-metrics`):
```
[the remark this answers is not in the document]
Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:918]: I want to describe a common issue where node level metrics are flowing but cluster level metrics are failing (only these ones). Probably I will just remove the `Only`, because even if all metrics are failing, the troubleshooting details for cluster-level ones would still be valid (although when all fails there are more chances for the cause to be at permissions, endpoint or connectivity issues)

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 6]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
```
Comment (eedugon) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:918]:
```suggestion
* Missing cluster-level metrics (provided by `kube-state-metrics`):
```

Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:918]: Better, @bmorelli25 ? :)

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 1 above)
Comment (eedugon) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:864]:
```suggestion
* https://github.com/kubernetes/kube-state-metrics[Kube-state-metrics] deployed as a `Deployment`
```

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 7]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+[source,sh]
+----
+kubectl kustomize https://github.com/elastic/elastic-agent/deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-standalone\?ref\=v8.15.3 | sed -e 's/JUFQSV9LRVkl/ZDAyNnZaSUJ3eWIwSUlCT0duRGs6Q1JfYmJoVFRUQktoN2dXTkd0FNMtdw==/g' -e "s/%ES_HOST%/https:\/\/7a912e8674a34086eacd0e3d615e6048.us-west2.gcp.elastic-cloud.com:443/g" -e "s/%ONBOARDING_ID%/db687358-2c1f-4ec9-86e0-8f1baa4912ed/g" -e "s/\(docker.elastic.co\/beats\/elastic-agent:\).*$/\18.15.3/g" -e "/{CA_TRUSTED}/c\ " > elastic_agent_installation_complete_manifest.yaml
+----
+
+The previous generates a local file named `elastic_agent_installation_complete_manifest.yaml`, which you can use for further analysis. It contains the complete set of resources required for the {agent} installation, including:
+
+* RBAC objects (`ServiceAccounts`, `Roles`, etc. )
+
+* `ConfigMaps` and `Secrets` for {agent} configuration.
```
Comment (eedugon) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:860]:
```suggestion
* `ConfigMaps` and `Secrets` for {agent} configuration
```

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:918]: resolving, as I've prepared a change related with this.

**Comment by eedugon** [comment]
@kilfoyle : I'm done with Brandon suggestions, only a tiny change is pending :)

Feel free to share your feedback and review. 

In my opinion we should probably end up creating a complete document focused on Agent on Kubernetes troubleshooting, but even though this might be a good start.

Also note that the section I'm adding is a bit different in nature as the other sections of the troubleshooting document because I didn't have a list of exact issues to report, and it's a bit more generic about the manifest installation with kustomize and possible issues.

If you want to add specific issues I would be more than happy to add content if you share, or if you believe we should structure this in a different way.

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 8]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+
+* RBAC objects (`ServiceAccounts`, `Roles`, etc.)
+
+* `ConfigMaps` and `Secrets` for {agent} configuration
+
+* {agent} Standalone deployed as a `DaemonSet`
+
+* https://github.com/kubernetes/kube-state-metrics[Kube-state-metrics] deployed as a `Deployment`
+
+The content of this file is equivalent to the one obtained when following the "<<running-on-kubernetes-standalone>>" document, with the exception of `kube-state-metrics`, which is not included in the other method.
```
Comment (kilfoyle) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:866]:
```suggestion
The content of this file is equivalent to what you'd obtain by following the <<running-on-kubernetes-standalone>> steps, with the exception that `kube-state-metrics` is not included in the standalone method.
```
Just a suggested rephrasing. One interesting thing about the Elastic docs is that we try to address the user directly, as "you".

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 9]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+
+* Some Kubernetes security mechanisms (like https://kubernetes.io/docs/concepts/security/pod-security-standards/[Pod Security Standards]) could cause part of the manifest to be rejected, as `hostNetwork` access and `hostPath` volumes are required.
+
+* If you already have an installation of `kube-state-metrics`, it could cause part of the manifest installation to fail or to update your existing resources without notice.
+
+[discrete]
+[[agent-kustomize-after]]
+=== Failures occurring within specific components after installation
+
+If the installation is correct and all resources are deployed, but data is not flowing as expected (e.g., you don't see any data on the *[Metrics Kubernetes] Cluster Overview* dashboard), check the following items:
```
Comment (kilfoyle) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:880]:
```suggestion
If the installation is correct and all resources are deployed, but data is not flowing as expected (for example, you don't see any data on the *[Metrics Kubernetes] Cluster Overview* dashboard), check the following items:
```
We avoid "e.g.", "i.e.", and any Latin terms, just as a way to make things a bit easier for readers less familiar with written English.

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 10]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+. <<agent-kustomize-manifest>>.
+. <<agent-kustomize-after>>.
+
+[discrete]
+[[agent-kustomize-manifest]]
+=== Problems related to the creation of objects within the manifest
+
+When troubleshooting installations performed with https://github.com/kubernetes-sigs/kustomize[Kustomize], it's good practice to inspect the output of the rendered manifest. To do this, take the installation command provided by Kibana Onboarding and replace the final part, `| kubectl apply -f-`, with a redirection to a local file. This allows for easier analysis of the rendered output.
+
+For example, the following command, originally provided by Kibana for an {agent} Standalone installation, has been modified to redirect the output for troubleshooting purposes:
```
Comment (kilfoyle) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:849]:
```suggestion
For example, the following command, originally provided by {kib} for an {agent} Standalone installation, has been modified to redirect the output for troubleshooting purposes:
```
 - `{kib}`  → Kibana
 - `{es}` → Elasticsearch

Most Elastic product names have variables. They're all defined in this [docs/shared/attributes.asciidoc ](https://github.com/elastic/docs/blob/88d54f34a0631de8a48d689b461d1e542d10caa7/shared/attributes.asciidoc#L219)file.

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 11]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+----
++
+[source,sh]
+----
+kubectl logs -n kube-system <name_of_kube-state-metrics_pod> 
+----
+
+Possible issues:
+
+* Connectivity, authorization, or authentication issues when connecting to Elasticsearch:
```
Comment (kilfoyle) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:912]:
```suggestion
* Connectivity, authorization, or authentication issues when connecting to {es}:
```

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 12]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
++
+Ensure the API Key and Elasticsearch destination endpoint used during the installation is correct and is reachable from within the Pods.
```
Comment (kilfoyle) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:914]:
```suggestion
* Connectivity, authorization, or authentication issues when connecting to {es}:
+
Ensure the API Key and {es} destination endpoint used during the installation is correct and is reachable from within the Pods.
```

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 13]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+. Determine which Pod owns the leadership `lease` in the cluster, with:
++
+[source,sh]
+----
+kubectl get lease -n kube-system elastic-agent-cluster-leader
+----
++
+. Check the logs of that Pod to see if there are errors when connecting to `kube-state-metrics` and if the `state_*` metrics are being sent.
++
+One way to check if `state_*` metrics are being delivered to Elasticsearch is to inspect log lines with the `"Non-zero metrics in the last 30s"` message and check the values of the "state_*" metricsets within the line, with something like:
```
Comment (kilfoyle) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:931]:
```suggestion
One way to check if `state_*` metrics are being delivered to {es} is to inspect log lines with the `"Non-zero metrics in the last 30s"` message and check the values of the "state_*" metricsets within the line, with something like:
```

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 14]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
++
+[source,sh]
+----
+kubectl logs -n kube-system elastic-agent-xxxx | grep "Non-zero metrics" | grep "state_"
+----
++
+If the previous command returns something like `"state_pod":{"events":213,"success":213}` for all `state_*` metricsets, it means the metrics are being delivered.
```
Comment (kilfoyle) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:938]:
```suggestion
If the previous command returns `"state_pod":{"events":213,"success":213}` or similar for all `state_*` metricsets, it means the metrics are being delivered.
```
(just to avoid repeating "something like")

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 15]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+. As a last resort, if you believe none of the Pods is acting as a leader, you can try deleting the `lease` to generate a new one:
++
+[source,sh]
+----
+kubectl delete lease -n kube-system elastic-agent-cluster-leader
+# wait a few seconds and check for the lease again
+kubectl get lease -n kube-system elastic-agent-cluster-leader
+----
+
+* Performance problems
```
Comment (kilfoyle) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:949]:
```suggestion
* Performance problems:
```
to match the previous bullets.

**Comment by kilfoyle** [comment]
@eedugon  I think this looks terrific! I added just a few super minor comments.

> In my opinion we should probably end up creating a complete document focused on Agent on Kubernetes troubleshooting, but even though this might be a good start.

Yeah, I agree that we eventually should break this out as a separate page, especially if we expand it with more info. For this PR, though, I think what you're adding is perfect.

I'm not aware of any other related content to add right now, but it's nice to have a good place for it.  :-)

**Comment by eedugon** [comment]
thanks a lot @kilfoyle for the comments and suggestions!!

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 9 above)
Comment (bmorelli25) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:880]: Oops. Thanks. That was actually my suggestion to add that. I didn't know we avoid them 😄

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 6 above)
Comment (bmorelli25) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:918]: Ahh, I see. Yes, this looks good!

**Review by bmorelli25 — APPROVED** [review]
Nice stuff! 🦕

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 13 above)
Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:931]: I knew it, but I didn't pay enough attention... :)
thanks!!

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 16]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+== Troubleshoot {agent} installation on Kubernetes, with Kustomize
+
+Potential issues during {agent} installation on Kubernetes can be categorized into two main areas:
+
+. <<agent-kustomize-manifest>>.
+. <<agent-kustomize-after>>.
+
+[discrete]
+[[agent-kustomize-manifest]]
+=== Problems related to the creation of objects within the manifest
```
Comment (gizas) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:845]:
```suggestion
=== Problems related to the creation of resources within the manifest
```

nit: Maybe resources sounds more k8s friendly?

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 17]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+----
+kubectl describe -n kube-system <name_of_elastic_agent_pod>
+----
+
+. Check the logs of elastic-agents and kube-state-metrics, and look for errors:
++
+[source,sh]
+----
+kubectl logs -n kube-system <name_of_elastic_agent_pod> 
+kubectl logs -n kube-system <name_of_elastic_agent_pod> | grep -i error
```
Comment (gizas) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:902]: What do you think sometimes warnings as well can help?

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 18]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+
+* Connectivity, authorization, or authentication issues when connecting to {es}:
++
+Ensure the API Key and {es} destination endpoint used during the installation is correct and is reachable from within the Pods.
++
+In an already installed system, the API Key is stored in a `Secret` named `elastic-agent-creds-<hash>`, and the endpoint is configured in the `ConfigMap` `elastic-agent-configs-<hash>`.
+
+* Only missing cluster-level metrics (provided by `kube-state-metrics`):
++
+These metrics (`state_*`) are retrieved by one of the Pods acting as `leader` (as described in <<running-on-kubernetes-standalone>>), so in order to troubleshoot that situation:
```
Comment (gizas) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:920]: I think that we understand what `(`state_*`)` means, but lets consider users that first time install our solution.

Maybe we need to describe a bit this. somthing like that metrics that bloneg to datastreams with prefix state_* etc ... and put a link to https://www.elastic.co/guide/en/elasticsearch/reference/current/data-streams.html

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 19]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+[[agent-kustomize-after]]
+=== Failures occurring within specific components after installation
+
+If the installation is correct and all resources are deployed, but data is not flowing as expected (for example, you don't see any data on the *[Metrics Kubernetes] Cluster Overview* dashboard), check the following items:
+
+. Check resources status and ensure they are all in a `Running` state:
++
+[source,sh]
+----
+kubectl get pods -n kube-system | grep elastic
```
Comment (gizas) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:886]: Maybe add a note that that the default namespace of installation is kube-system for all kustomize templates

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 20]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
+These metrics (`state_*`) are retrieved by one of the Pods acting as `leader` (as described in <<running-on-kubernetes-standalone>>), so in order to troubleshoot that situation:
++
+. Determine which Pod owns the leadership `lease` in the cluster, with:
++
+[source,sh]
+----
+kubectl get lease -n kube-system elastic-agent-cluster-leader
+----
++
+. Check the logs of that Pod to see if there are errors when connecting to `kube-state-metrics` and if the `state_*` metrics are being sent.
```
Comment (gizas) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:929]: The kube-state-metrics and the agent should co-exit in same namespace in order the communicate. If no and users change this ,then furhter policy updates are needed. 

Maybe we can add a note that ageent and ksm are installed in same namespace

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc [hunk 21]
```diff
@@ -830,3 +831,129 @@ Error: error loading agent config: error loading raw config: fail to read config
...
++
+Ensure the API Key and {es} destination endpoint used during the installation is correct and is reachable from within the Pods.
++
+In an already installed system, the API Key is stored in a `Secret` named `elastic-agent-creds-<hash>`, and the endpoint is configured in the `ConfigMap` `elastic-agent-configs-<hash>`.
+
+* Only missing cluster-level metrics (provided by `kube-state-metrics`):
++
+These metrics (`state_*`) are retrieved by one of the Pods acting as `leader` (as described in <<running-on-kubernetes-standalone>>), so in order to troubleshoot that situation:
++
+. Determine which Pod owns the leadership `lease` in the cluster, with:
```
Comment (gizas) [inline docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:922]:
```suggestion
. Determine which Pod owns the [leadership](https://www.elastic.co/guide/en/fleet/current/kubernetes_leaderelection-provider.html#:~:text=Provides%20the%20option%20to%20enable,Elastic%20Agent%20holds%20the%20leadership.) `lease` in the cluster, with:
```

**Review by kilfoyle — DISMISSED** [review]
LGTM! 🚀

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 18 above)
Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:920]: totally agreed! I'll work on that, thanks!

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 20 above)
Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:929]: Perfect!

Would you like to include the needed steps to install the agent pointing to a `ksm` deployed in a different namespace? If you share the instructions I could also add it in case the customer already has a ksm and doesn't want our manifest to deploy it.

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 17 above)
Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:902]: yes, I will add that too, not sure if with both in the same grep command or better in a separate line to ensure command compatibility.

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 16 above)
Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:845]: I have always considered and and referred to them as `kubernetes resources` too, and I see they are widely referenced as `resources`.

But some time ago, when looking at the official docs they are actually presented as objects: https://kubernetes.io/docs/concepts/overview/working-with-objects/

Also in these current blog posts the authors show a clear distinction:
- https://devopscube.com/kubernetes-objects-resources/
- https://iximiuz.com/en/posts/kubernetes-api-structure-and-terminology/

So, I think the right way to call them is `objects` (although it still feels strange to me).

Anyway, happy to change it to `resources` if you prefer. This is a topic I like, because in all the training I've been delivering for years I've always talked about k8s resources, and now I'm trying to talk about `objects` when referring to the persistent entities that represent the state of the cluster / records of intent (like `Pods`, `Deployments`, etc).

In summary:
- IMO a manifest uses and defines API resources that end up creating objects.
- A user should take care and check the objects related with the manifests they applied.

Comment (gizas) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:845]: You covered all my concerns :) Thanks for the detailed reply. I truly learned a lot here

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 17 above)
Comment (gizas) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:902]: Sure whatever you want

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 20 above)
Comment (gizas) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:929]: One option is: To manually change the namespace in the kustomize templates of agent in order the namespace to be the smae
Or I guess to change the url of the https://github.com/elastic/elastic-agent/blob/main/deploy/kubernetes/elastic-agent-standalone/elastic-agent-standalone-ksm-statefulset-configmap.yaml#L50 ksm endpoint if still agent is installed in kube-system

But needs some testing
I would say that this complicates more the things if we add so many details now

Comment (gizas) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:929]: Maybe this can be a separate doc

**Review by gizas — DISMISSED** [review]

File: docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc (hunk 18 above)
Comment (eedugon) [reply docs/en/ingest-management/troubleshooting/troubleshooting.asciidoc:920]: I have updated it to:

> Cluster level metrics are provided in {ref}/data-streams.html[data streams] prefixed with `state_`, by one of the Pods acting as `leader` (as described in <<running-on-kubernetes-standalone>>), so in order to troubleshoot that situation:

I'll commit it soon.

**Review by kilfoyle — APPROVED** [review]

**Review by gizas — APPROVED** [review]
